src/pages/ProductDetailsPage.jsx

```
@@ -93,24 +92,28 @@ export default function ProductDetailsPage() {
+    if (productError || variantsError) {
+        return <ErrorDisplay error={productError || variantsError} />;
     }
 
-    // Determine current price (from variation or base product)
-    const currentPrice = selectedVariation?.price || product.price;
-    const salePrice = product.salePrice;
-    const hasDiscount = salePrice && salePrice < currentPrice;
+    // Determine current price (from variant or base product)
+    const currentPrice = selectedVariant?.price || product.price;
+    const hasDiscount = product.salePrice && product.salePrice < currentPrice;
+    const displayPrice = hasDiscount ? product.salePrice : currentPrice;
```

ESLint

[error] 'displayPrice' is assigned a value but never used. Allowed unused vars must match /^[A-Z_]/u.

(no-unused-vars)